ui/StatusQ/src/imageutils.cpp

```diff
@@ -0,0 +1,66 @@
+#include "StatusQ/imageutils.h"
+
+#include <QDebug>
+#include <QEventLoop>
+#include <QImage>
+#include <QNetworkAccessManager>
+#include <QNetworkReply>
+#include <QUrl>
+#include <QUuid>
+
+namespace {
+constexpr auto kBase64JPGPrefix = "data:image/jpeg;base64,";
+}
+
+ImageUtils::ImageUtils(QObject *parent)
+    : QObject(parent)
+{}
+
+QString ImageUtils::resizeImage(const QString &imagePathOrData, int maxSize, const QString& tmpDirPath) const
+{
+    QImage img;
+    bool loadResult = false;
+
+    // load the contents
+    if (imagePathOrData.startsWith(kBase64JPGPrefix)) { // binary BLOB
+        loadResult = img.loadFromData(QByteArray::fromBase64(imagePathOrData.mid(qstrlen(kBase64JPGPrefix)).toLatin1())); // strip the prefix, decode from b64
+    } else { // local file or URL
+        const auto localFileOrUrl = QUrl::fromUserInput(imagePathOrData); // accept both "file:/foo/bar" and "/foo/bar"
+        if (localFileOrUrl.isLocalFile()) {
+            loadResult = img.load(localFileOrUrl.toLocalFile());
+        } else {
+            QEventLoop loop;
```

Comment:
I think it's kind of workaround that should be avoided. It makes false impression that's synchronous function but actually it's not. It may also cause hard to detect problems when another event clears the content of that invocation (e.g. Loader where it was called becomes inactive).

I think we should strongly consider approach used by built-in qml types, e.g. https://doc.qt.io/qt-6/qml-qtquick-item.html#grabToImage-method

Reply:
Yeah I agree here... I don't like this either. But we don't want to grab anything here, rather download the file from a remote URL

Reply:
I'll drop the support for remote URLs here